### ISSUE TITLE
Please output Makefile-style depfiles for ninja and make

### ISSUE THREAD

**Issue by ximion** [ldc-developers/ldc#1802]
GCC and other compilers emit dependency information as Make-style depfiles, allowing Make and make-replacements like Ninja to recompile dependencies as soon as some unit higher up in the dependency chain has been changed.

This is tremendously useful, especially in D, where descending units have to be rebuild e.g. on a template change or function signature change.

Unfortunately, LDCs `-deps=` flag makes it output a custom format, which makes adding LDC to build systems using Ninja and Make harder than it needs to be.

The Makefile-syntax used by GCC and Clang is the de-facto standard and changing all other tools to support LDC will be impossible.

See https://ninja-build.org/manual.html#_depfile for a bit more information on depfiles. Unfortunately, no format documentation exists, but the format is simply a Makefile without variables in the form of:

``` Make
the_generated_file.o:
        source1.d source2.d source3.d
```

Ninja's tests also contain a bit of information: https://github.com/ninja-build/ninja/blob/master/src/depfile_parser_test.cc

Outputting both formats would be helpful, GDC has `-fdeps=` and `-fmake-deps=` for that.
Cheers,
    Matthias

**Comment by kinke** [comment]
Certainly interesting. A ddmd feature request for this `-make-deps=` (possibly upstreaming GDC's solution) would be more appropriate though, so that all D compilers support it.

**Comment by ximion** [comment]
I was going to file a request against DMD anyway, so yeah, that's likely a good idea :)

**Comment by ximion** [comment]
Reported as https://issues.dlang.org/show_bug.cgi?id=16746

**Comment by ximion** [comment]
FTR, this feature is in a PR for DMD that doesn't get reviewed for a while: https://github.com/dlang/dmd/pull/6961

**Comment by kinke** [comment]
This was fixed with `-makedeps` introduced in D 2.095.
